Fix working dir checkout (#4879)
ocaml/opam: master_changes.md

## Pin
  * Switch the default version when undefined from ~dev to dev [#4949 @kit-ty-kate]

>   * Don't read opam file in source for version pin [#4931 @rjbou - #4608]

Author: ```suggestion
```
add git: clean working dir before looking for diff in changes

  * [BUG] Fix repin of locked pins when there is no change in lock file [#5079 @rjbou - fix #4313]
  * [BUG] Fix `opam install ./file.opam` lock pinning [#5148 @rjbou - fix #4313]
  * [BUG] Fix origin opam file retrieval when opam originate from locked file [#5079 @rjbou - fix #4936]

>   * [BUG] When reinstalling a package that has a dirty source, if uncommitted changes are the same than remotely committed, opam consider that it is up to date and nothing is updated [4879 @rjbou]

Reviewer: ```suggestion
  * [BUG] When reinstalling a package that has a dirty source, if uncommitted changes are the same than the ones stored in opam's cache, opam consider that it is up to date and nothing is updated [4879 @rjbou]
```

  * Add `?subpath` to `OpamRepository.fetch_dev_packages`, `OpamVCS.is_up_to_date` and vcs specific functions in `OpamDarcs`, `OpamHG`, and `OpamGit` [#4876 @rjbou]
  * `OpamRepositoryConfig.E`: add `curl_t` and `fetch_t` to get their respective environement vairbales value dynamically, without lazyness. It is used in `opamClient.InitDefaults`, that can be called at topelevel [#5111 @rjbou]
  * `OpamRepository.update`: Return a change state result of the repo update [#5043 @Armael]

>   * `OpamGit.fetch`: clean with `reset --hard` before fetching [#4879 @rjbou]

Reviewer: this isn’t the case in the PR anymore.

Reviewer: ```suggestion
  * `OpamVCS.VCS`: add a `clean` function to the interface clearing all the uncommited files [#4879 @rjbou]
```

Author: Well spotted, thanks!
